src/bitcoin/Address.mo

```diff
@@ -1,26 +1,64 @@
 import P2pkh "./P2pkh";
+import P2tr "./P2tr";
 import Script "./Script";
+import Segwit "../Segwit";
 import Types "./Types";
 import Result "mo:base/Result";
 
 module {
+  public func addressFromText(address : Text) : Result.Result<Types.Address, Text> {
+    switch (Segwit.decode(address)) {
+      case (#ok _) {
+        return #ok(#p2tr_key(address));
+      };
+      case (_) {};
+    };
+
+    switch (P2pkh.decodeAddress(address)) {
+      case (#ok _) {
+        return #ok(#p2pkh(address));
+      };
+      case (_) {};
+    };
+
+    #err("Failed to decode address " # address);
```

Comment:
Other formats are currently not supported? At the very least, we should have support for `P2WPKH`, which is used quite a lot, right?

Reply:
This would make sense in general. Here, we focused more on what we need to implement to show how to use Schnorr signatures with Bitcoin. P2PKH was implemented long ago and already at that point it did make sense to have P2WPKH. So I would guess that was a matter of prioritization. We will discuss this internally.